extensions/resource-deployment/src/ui/deploymentDialog.ts

```diff
@@ -46,6 +48,22 @@ export class DeploymentDialog extends DialogBase {
 				);
 
 				const form = formBuilder.withLayout({ width: '100%' }).component();
+				const self = this;
+				this._dialogObject.registerCloseValidator(() => {
+					const messages: string[] = [];
+					self.validators.forEach(validator => {
+						const result = validator();
+						if (!result.valid) {
+							messages.push(result.message);
+						}
+					});
+					if (messages.length > 0) {
+						self._dialogObject.message = { level: azdata.window.MessageLevel.Error, text: messages.join('\n') };
```

Comment:
On Windows, multiple messages show on different lines, correct?

Reply:
yes, tested on windows and mac.